site/source/docs/api_reference/val.h.rst

```diff
@@ -48,7 +46,13 @@ Guide material for this class can be found in :ref:`embind-val-guide`.
     }
 
   See :ref:`embind-val-guide` for other examples.
+  
+
+  .. warning:: JavaScript values can't be shared across threads, so neither can ``val`` instances that bind them. For example, if you want to cache some JavaScript global as a ``val``, you need to retrieve that global separately in each thread and bind to thread-local ``val`` bindings:
```

Comment:
This looks a lot better, but it still confuses me. It's not the same global in each thread. But I think I see what you are getting at now.
```suggestion
  .. warning:: JavaScript values can't be shared across threads, so neither can ``val`` instances that bind them. For example, if you want to cache some JavaScript global as a ``val``, you need to retrieve that global (that is, the global of that name, in that thread's JS context) separately in each thread and bind to thread-local ``val`` bindings:
```

Reply:
If you think slightly more verbose explanation is useful, how about this:

```suggestion
  .. warning:: JavaScript values can't be shared across threads, so neither can ``val`` instances that bind them.
    
    For example, if you want to cache some JavaScript global as a ``val``, you need to retrieve and bind separate instances of that global by its name in each thread.
    The easiest way to do this is with a ``thread_local`` declaration:
```